### PR TITLE
Add verifyProofHashForStateTree() util function (with verifyProofHashForRadixTree())

### PR DESCRIPTION
**Pull request by platfowner**
Change summary:
- Add verifyProofHashForStateTree() util function (with verifyProofHashForRadixTree())
- Add verifySnapshot.js tool
- Update proof hash tool sample files
- Update verifyBlock.js tool to the best (still not working)

Related issues:
- https://github.com/ainblockchain/ain-blockchain/issues/649
- https://github.com/ainblockchain/ain-blockchain/issues/713

### CODE REVIEW

File: db/radix-node.js [hunk 1]
```diff
@@ -489,6 +492,47 @@ class RadixNode {
...
+      const childStateNode = this.getChildStateNode();
+      const proofStateLabel = StateInfoProperties.STATE_LABEL_PREFIX + childStateNode.getLabel();
+      const childStateVerif =
+          verifyProofHashForStateTree(childStateNode, [...curLabels, proofStateLabel]);
+      if (childStateVerif.isVerified !== true) {
+        return childStateVerif;
+      }
+      preimage += childStateNode.getProofHash();
+    }
+    preimage += LIGHTWEIGHT ? '' : `${HASH_DELIMITER}`;
```
Comment (minsulee2) [inline db/radix-node.js:508]: What about ${HASH_DELIMITER} => HASH_DELIMITER ?

File: tools/proof-hash/verifyBlock.js [hunk 2]
```diff
@@ -1,53 +1,77 @@
...
+} = require('../../common/constants');
+const BlockchainNode = require('../../node');
 const Consensus = require('../../consensus');
+const ConsensusUtil = require('../../consensus/consensus-util');
+
+async function loadBlocks(blockFileList) {
+  const blockList = [];
+  for (let i = 0; i < blockFileList.length; i++) {
+    const blockFile = blockFileList[i];
+    console.log(`\n<< [${i + 1}]: ${blockFile} >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>`);
```
Comment (minsulee2) [inline tools/proof-hash/verifyBlock.js:15]: Do we keep console for here?!

**Review by minsulee2 — APPROVED** [review]
LGTM with minor comments!

File: db/state-util.js [hunk 3]
```diff
@@ -839,6 +835,34 @@ function verifyStateInfoForStateTree(stateTree) {
   return true;
 }
 
+function verifyProofHashForStateTree(stateTree, curLabels = []) {
+  const curPath = CommonUtil.formatPath(curLabels);
+  if (stateTree.getIsLeaf()) {
+    const proofHashComputed = LIGHTWEIGHT ?
```
Comment (liayoo) [inline db/state-util.js:841]: ditto

File: node/index.js [hunk 4]
```diff
@@ -42,13 +42,13 @@ const BlockPool = require('../block-pool');
 const ConsensusUtil = require('../consensus/consensus-util');
 
 class BlockchainNode {
-  constructor() {
+  constructor(account = null) {
```
Comment (liayoo) [inline node/index.js:45]: nice~ 👍

File: tools/proof-hash/verifyBlock.js [hunk 5]
```diff
@@ -1,53 +1,77 @@
...
+
+    const block = blockList[i];
+    const nextBlock = i + 1 < blockList.length ? blockList[i + 1] : null;
+    proposalTx = nextBlock ? ConsensusUtil.filterProposalFromVotes(nextBlock.last_votes) : null;
+
+    console.log(`\n* Executing block on db...`);
     try {
-      Consensus.validateAndExecuteBlockOnDb(block, snapDb, prevBlock);
-      prevBlock = block;
+      Consensus.validateAndExecuteBlockOnDb(block, node, 'verifyBlock', proposalTx);
```
Comment (liayoo) [inline tools/proof-hash/verifyBlock.js:69]: thanks for updating this!

**Review by liayoo — APPROVED** [review]
LGTM with some comments. Thanks!

File: tools/proof-hash/verifyBlock.js (hunk 2 above)
Comment (platfowner) [reply tools/proof-hash/verifyBlock.js]: Removed, thanks.

File: db/radix-node.js (hunk 1 above)
Comment (platfowner) [reply db/radix-node.js]: Updated, thanks.

[the remark this answers is not in the document]
Comment (platfowner) [reply db/radix-node.js]: Right, just remove LIGHTWEIGHT flag related parts for simplicity.

File: db/state-util.js (hunk 3 above)
Comment (platfowner) [reply db/state-util.js]: ditto

File: node/index.js (hunk 4 above)
Comment (platfowner) [reply node/index.js:45]: thanks!

File: tools/proof-hash/verifyBlock.js (hunk 5 above)
Comment (platfowner) [reply tools/proof-hash/verifyBlock.js:103]: tried to update the code, but the block execution part is not working properly yet. just added a todo for further work.

**Comment by platfowner** [comment]
Thanks for the fast review! Now merging..

**Pull request merged**